### ISSUE TITLE
Excel can't open file because some content is unreadable

### ISSUE THREAD

**Issue by yjukaku** [straydogstudio/axlsx_rails#41]
Trying to use this gem for the first time, I spent about 2 hours trying to solve the error:

`Excel could not open payments.xlsx because some content is unreadable. Do you want to open and repair this workbook?`

The problem was that I missed the documentation stating to use `wb = xlsx_package.workbook` instead of `p = Axlsx::Package.new`.

Can we add this to the `Troubleshooting` section of the README?

**Comment by straydogstudio** [comment]
@yjukaku I definitely will. Thanks for providing the feedback.

**Comment by DaleGardner** [comment]
![giant_thumb](https://cloud.githubusercontent.com/assets/3970939/9888511/80c09bf4-5bc3-11e5-8525-6b5951e18255.png)

**Comment by moranpro** [comment]
I just encountered the same problem. @straydogstudio , is there a fix? I am using Excel for Mac 2011, Version 14.6.0 on a MacBook Pro running Yosemite. I have potentially lost many hours of work on one of two sheets in a .xlsx workbook. Any help is greatly appreciated. Thanks in advance.

**Comment by straydogstudio** [comment]
Assuming you have not missed using the passed in `xlsx_package` mentioned above, have you tried using `render xlsx: 'whatever', layout: false`? 

Sometimes, depending on what you are trying to open it with, using shared strings is also important. 

It may be worthwhile making a script that you run with Rails runner to create the file using Axlsx without axlsx_rails. That will show if it is specific to axlsx or axlsx_rails. 

I will get back to you tomorrow or later tonight if I can.

**Comment by moranpro** [comment]
Thanks for the reply. The above is Greek to me so, no, I have not tried using the string mentioned above. Also, I do not know Rails runner.

Unfortunately, I am pretty married to Mac format now, as much as i wish I wasn't, and am not nearly as conversant on that platform as I was on Windows - at least the earlier versions.

**Comment by yjukaku** [comment]
@moranpro @moranpro Just to be clear, are you writing software using Ruby?

**Comment by moranpro** [comment]
No. I had constructed a couple of interior drawings using Excel: a plan drawing on one sheet and an isometric on the other. The file will not open without being "repaired", which loses the second sheet. I have not saved the file following repair, so the original, corrupted one still exists.

**Comment by straydogstudio** [comment]
@moranpro Sorry but I can't help you. It looks like you have a corrupted file. If you use Time Machine to backup your files (built into OSX) then you could easily recover a previous version. If not, you should get an external drive and use it. Well worth it.

**Comment by moranpro** [comment]
OK. I will check. I do back up regularly, but I am not sure that there was a backup done between the time I created the second sheet and the time I attempted to reopen it. Fingers crossed.

Time Machine scares me because one never really knows whether the backups are OK. I had to change out my HD a few months ago and the Time Machine backups, which were supposed to be complete, were not even close. Fortunately, I had manually copied critical data files to another hard drive prior to replacing the internal drive.
